applications/tari_app_utilities/src/commands.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::command_parser::{ParsedArgument, ParsedCommand};
+use chrono::{DateTime, Utc};
+
+use log::*;
+use std::{str::FromStr, time::Duration};
+use strum_macros::{Display, EnumIter, EnumString};
+use tari_common::GlobalConfig;
+use tari_core::transactions::tari_amount::uT;
+use tari_wallet::{
+    transaction_service::{
+        error::TransactionServiceError,
+        handle::{TransactionEvent, TransactionServiceHandle},
+    },
+    WalletThreadSafe,
+};
+use thiserror::Error;
+use tokio::{
+    runtime::Handle,
+    stream::StreamExt,
+    task::JoinError,
+    time::{delay_for, timeout},
+};
+
+/// Enum representing commands used by the wallet
```

Comment:
Sure that makes sense, I wanted to refactor the base node to utilise the same parsing and executing logic, but my rust-fu isn't quite there yet :) Happy to move it into a console wallet module

Reply:
I believe in the end the wallet functionality will be completely removed from the Base Node and only live in the wallet app which is primarily why I believe this belongs in that crate. If this parsing ends up being used in a second app then I think it can move back here though if possible I would want to keep the dependency on the Transaction Service Handles out of here as that implies this is more than just parsing.

Reply:
That makes sense, thanks @philipr-za